tools/common/Rewriter.cs

```diff
@@ -220,9 +225,12 @@ void PatchMethods (TypeDefinition cl, FieldDefinition classPtr, FieldDefinition
 
 		void PatchMethod (MethodDefinition method, FieldDefinition classPtr, FieldDefinition classPtrField)
 		{
-			var il = method.Body.GetILProcessor ();
-			for (int i = 0; i < method.Body.Instructions.Count; i++) {
-				var instr = method.Body.Instructions [i];
+			var body = method.Body;
+			if (body is null)
+				return;
+			var il = body.GetILProcessor ();
+			for (int i = 0; i < body.Instructions.Count; i++) {
+				var instr = body.Instructions [i];
```

Comment:
@haritha-mohan This will create an invalid operation exception. Body.Instructions is a collection type and later on we modify the contents of the collection which invalidates the enumerator.

Reply:
@stephen-hawley ah fair enough, the [source code](https://github.com/jbevain/cecil/blob/56d4409b8a0165830565c6e3f96f41bead2c418b/Mono.Cecil.Cil/ILProcessor.cs#L250) from cecil shows how the replace method does modify the collection so the index is necessary. thanks for clarifying!